episodes/26-log-regression.Rmd

```diff
@@ -0,0 +1,333 @@
+---
+title: "Logistic Regression"
+teaching: 60
+exercises: 60
+source: Rmd
+---
+
+::::::::::::::::::::::::::::::::::::::: objectives
+
+- To be able to construct regression models for a binary outcome
+- To be able to calculate predicted variables and residuals 
+- To be able to present model outcomes using Broom
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::: questions
+
+- How can I identify factors for antibiotic resistance?
+- How can I check the validity of model?
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Content
+
+-   Exploratory Data Analysis
+-   Model Creation and Estimation
+-   Reporting the Logistic Regression Results
+
+## Data
+
+```{r libraries, message=FALSE, warning=FALSE}
+
+# We will need these libraries and this data later.
+library(aod)
+library(broom)
+library(ggplot2)
+library(lubridate)
+library(odds.n.ends)
+library(tidyverse)
+
+amrData <- read.csv("data/dig_health_hub_amr.csv")
+
+```
+The data used in this episode was provided by Simon Thelwall from the UKHSA. 
+It has been created to represent the sort of data that might be obtained from the Second Generation Surveillance System (SGSS).
+The data has 100,000 rows of 12 variables.
+
+## Exploratory Data Analysis
+
+We can preview our data by using 'head':
+
+```{r}
+
+head(amrData)
+
+```
+We can also request information about variable names and data types:
+```{r}
+
+sapply(amrData, class)
+
+```
+
+We can see that dates are currently stored as the char data type. We also do not know the age of the participant when the sample was taken.
+
+```{r}
+
+# Calculate age (in years) as of their last birthday and add as an additional variable to our data.
+
+amrData <- amrData %>% 
+  mutate(
+    age_years_sd = (dob %--% spec_date) %/% years(1)
```

Comment:
Are the `%--%` and `%/%` syntax bits intentional? I've never seen this kind of index before, do we cover this in one of the previous episodes?

Reply:
Comment added to explain syntax and that it for lubridate